### ISSUE TITLE
Does not find Windows games downloaded with Itch app

### ISSUE THREAD

**Issue by Hatomike** [PhilipK/BoilR#95]
I installed some games from the Itch app on my Deck and noticed that BoilR does not find windows games.
Here is the list of the game I've tested, all of them have been launched at least once from the itch app and are working:

CrossCode (Linux) - DETECTED
SkateBIRD (Linux) - DETECTED
Baba Is You (Linux) - DETECTED
Moon Hunters (Windows)
Cloud Gardens (Windows)
INMOST (Windows)
BloodbornePSX (Windows)

If you want to try it out, you need to install Wine before Itch allow you to download windows games.
BloodbornePSX is also free if you do not have any windows games in your itch io library.

**Comment by PhilipK** [comment]
Thanks for reporting this @Hatomike 
I didn't even know that you could install windows games with Itch on Linux.

I will check out why windows games are not found.

**Comment by PhilipK** [comment]
So I can confirm that this is a bug.
The thing is, for some reason is that Itch normally writes the base path (install folder) for a game and then a list of "candidate" files to launch for a game, so for a windows game that should be the exe file to launch.
That is how BoilR finds the linux files to launch.

But for the windows games, the candidate list is "null".
Now it does add a receipt.json file in the install folder, while that file contains a list of all the files installed with the game, it does not state which one should be launched.

I have looked into the itch app, since it is opensource.
Its written in typescript and go, and while I know both languages, the code is ... a bit confusing :-)
I have not been able to find where the itch app handles windows games installed in linux yet.

**Comment by cherokee93** [comment]
Thanks for trying. If the native Linux version of itch works with your tool, it will get eventually fixed to support flatpak wine used on the Deck

**Comment by PhilipK** [comment]
I have tried itch.io on linux (arch and manjaro) and on windows and it works for me there.
If it doesn't work for you on windows that is a new bug then.

**Comment by cherokee93** [comment]
> I have tried itch.io on linux (arch and manjaro) and on windows and it works for me there. If it doesn't work for you ( @cherokee93 ) on windows that is a new bug then.

No, i'm on Steam Deck running windows itch.io in lutris. So is it working? I'll tinker around with the newest release and see

**Comment by PhilipK** [comment]
@cherokee93 if you have installed the windows version of itch, using proton, then all the internal paths that itch uses will be related to its "sandboxed" C: drive. BoilR will be able to parse the itch databse, but when it checks if the files actually exists it won't know to prepend the sandbox path.

In short, it won't work right now, but could be fixable.

Wouldn't it be better to just get the Linux version working?

**Comment by cherokee93** [comment]
> @cherokee93 if you have installed the windows version of itch, using proton, then all the internal paths that itch uses will be related to its "sandboxed" C: drive. BoilR will be able to parse the itch databse, but when it checks if the files actually exists it won't know to prepend the sandbox path.
> 
Yes but I used wine complete path, and inside that path the database files are present:
![image](https://user-images.githubusercontent.com/12370126/166158076-1e5bf061-2a03-4759-b6b8-c2b5888da671.png)



> Wouldn't it be better to just get the Linux version working?

Absolutely, the problem is that the linux version of itch.io didn't work with wine flatpak or proton version provided by SteamOS weeks ago when I tried.  It only displayed linux games and not windows ones

**Comment by cherokee93** [comment]
Ok I made it work, it need games to be in the default folder but for linux games it works. Awesome! 😊
Now I wait and hope they will fix their client (to create db entries?) for windows games

**Comment by Hatomike** [comment]
Just for confirmation.
I am using the native itch app and I installed Wine using pacman (after disabling read only).

As stated, Itch try to detect if Wine is installed before allowing the download of windows games but it does not work with the flatpak version of Wine.

Even if games can be launched with Wine, I still add the .exe as a non steam game to use Proton.
(It would be great if Itch could allow to bypass this behavior with a simple on/off switch)


I should have clarified all of this in my first message. Sorry.
Thanks for having a look at fixing this !

**Comment by Ugzuzg** [comment]
For a similar use-case [the developer suggested](https://github.com/itchio/itch/issues/2061#issuecomment-421055490) to launch a [butler daemon](https://docs.itch.ovh/butlerd/master/#/) and then sumbit a [launch](https://docs.itch.ovh/butlerd/master/#/?id=launch-category) request to it.
